services/consensusalgo/benchmarkconsensus/test/leader_test.go

```diff
@@ -2,19 +2,15 @@ package test
 
 import (
 	"context"
-	"github.com/orbs-network/go-mock"
 	"github.com/orbs-network/orbs-network-go/test"
 	"testing"
 )
 
 func TestLeaderCreatesBlocks(t *testing.T) {
 	test.WithContext(func(ctx context.Context) {
 		h := newHarness(true)
-		h.consensusContext.Reset().When("RequestNewTransactionsBlock", mock.Any).Return(nil, nil).AtLeast(1)
+		h.expectBlockCreation()
```

Comment:
now it seems like you pushed too much behavior into the driver and it's no longer clear from reading the test what exactly is the expected behavior

Reply:
Maybe I should return the mock function like you suggested in the last comment and this will move this behavior back to the test